Duplicati/Library/AutoUpdater/UpdaterManager.cs

```diff
@@ -439,15 +447,15 @@ public static bool DownloadUpdate(UpdateInfo version, PackageEntry package, stri
                             if (progress != null)
                                 cb = (s) => { progress(Math.Min(1.0, Math.Max(0.0, (double)s / package.Length))); };
 
-                            var wreq = (System.Net.HttpWebRequest)System.Net.WebRequest.Create(url);
-                            wreq.UserAgent = string.Format("{0} v{1}", APPNAME, SelfVersion.Version);
-                            wreq.Headers.Add("X-Install-ID", InstallID);
+                            using var request = new HttpRequestMessage(HttpMethod.Get, url);
+
+                            request.Headers.Add(System.Net.HttpRequestHeader.UserAgent.ToString(), string.Format("{0} v{1}", APPNAME, SelfVersion.Version));
+                            request.Headers.Add("X-Install-ID", InstallID);
+
+                            using var timeoutToken = new CancellationTokenSource();
+                            timeoutToken.CancelAfter(HttpContextSettings.ReadWriteTimeout);

```

Comment:
This should be `HttpContextSettings.OperationTimeout` as that is the overall operation timeout.